config/src/main/java/org/springframework/security/config/annotation/web/configurers/HeadersConfigurer.java

```diff
@@ -473,6 +479,168 @@ private FrameOptionsConfig enable() {
+		 */
+		public HpkpConfig withPins(Map<String, String> pins) {
+			writer.setPins(pins);
+			return this;
+		}
+
+		/**
+		 * <p>
+		 * Adds a value for the pin- directive of the Public-Key-Pins header.
+		 * </p>
+		 *
+		 * <p>
+		 * The pin directive specifies a way for web host operators to indicate
+		 * a cryptographic identity that should be bound to a given web host.
+		 * See <a href="http://tools.ietf.org/html/rfc7469#section-2.1.1">Section 2.1.1</a> for additional details.
+		 * </p>
+		 *
+		 * @param pin the base64-encoded SPKI fingerprint.
+		 * @param algorithm the cryptographic hash algorithm.
+		 * @throws IllegalArgumentException if algorithm or pin is null
+		 */
+		public HpkpConfig addPin(String pin, String algorithm) {
+			writer.addPin(pin, algorithm);
+			return this;
+		}
+
+		/**
+		 * <p>
+		 * Sets the value (in seconds) for the max-age directive of the Public-Key-Pins header.
+		 * The default is 60 days.
+		 * </p>
+		 *
+		 * <p>
+		 * This instructs browsers how long they should regard the host (from whom the message was received)
+		 * as a known pinned host. See <a href="http://tools.ietf.org/html/rfc7469#section-2.1.2">Section
+		 * 2.1.2</a> for additional details.
+		 * </p>
+		 *
+		 * @param maxAgeInSeconds the maximum amount of time (in seconds) to regard the host
+		 * as a known pinned host.
+		 * @throws IllegalArgumentException if maxAgeInSeconds is negative
+		 */
+		public HpkpConfig maxAgeInSeconds(long maxAgeInSeconds) {
+			writer.setMaxAgeInSeconds(maxAgeInSeconds);
+			return this;
+		}
+
+		/**
+		 * <p>
+		 * If true, the pinning policy applies to this pinned host as well as any subdomains
+		 * of the host's domain name. The default is false.
+		 * </p>
+		 *
+		 * <p>
+		 * See <a href="http://tools.ietf.org/html/rfc7469#section-2.1.3">Section 2.1.3</a>
+		 * for additional details.
+		 * </p>
+		 *
+		 * @param includeSubDomains true to include subdomains, else false
+		 */
+		public HpkpConfig includeSubDomains(boolean includeSubDomains) {
+			writer.setIncludeSubDomains(includeSubDomains);
+			return this;
+		}
+
+		/**
+		 * <p>
+		 * If true, the browser should not terminate the connection with the server. The default is true.
+		 * </p>
+		 *
+		 * <p>
+		 * See <a href="http://tools.ietf.org/html/rfc7469#section-2.1">Section 2.1</a>
+		 * for additional details.
+		 * </p>
+		 *
+		 * @param reportOnly true to report only, else false
+		 */
+		public HpkpConfig reportOnly(boolean reportOnly) {
+			writer.setReportOnly(reportOnly);
+			return this;
+		}
+
+		/**
+		 * <p>
+		 * Sets the URI to which the browser should report pin validation failures.
+		 * </p>
+		 *
+		 * <p>
+		 * See <a href="http://tools.ietf.org/html/rfc7469#section-2.1.4">Section 2.1.4</a>
+		 * for additional details.
+		 * </p>
+		 *
+		 * @param reportUri the URI where the browser should send the report to.
+		 */
+		public HpkpConfig reportUri(URI reportUri) {
```

Comment:
Should we also overload `.setReportUri(URI reportUri)`in `HPKPHeaderWriter` then?